AIM/AIM/Data/Gateways/XMLOnAirInfoGateway.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  XMLOnAirInfoGateway.swift
+//  AIM
+//
+//  Created by Oskar Szydlowski on 01.03.2018.
+//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
+//
+
+import Foundation
+
+typealias OnAirInfoCompletion = (OnAirInfo) -> ()
+
+class XMLOnAirInfoGateway: NSObject, XMLParserDelegate {
+    
+    private let onAirInfoFeedURL = URL(string: "http://aim.appdata.abc.net.au.edgesuite.net/data/abc/triplej/onair.xml")!
+    private var parser: XMLParser
+    
+    private var stationDisplayTime: String?
+    private var stationImageURL: URL?
+    private var stationInfo: StationInfo?
+    private var tracks = [Track]()
+    private var track: Track?
+    
+    private var parserCompletion: OnAirInfoCompletion?
+    
+    override init() {
+        self.parser = XMLParser(contentsOf: onAirInfoFeedURL)!
+        super.init()
+        parser.delegate = self
+    }
+    
+    func parse(completion: @escaping OnAirInfoCompletion) {
+        parserCompletion = completion
+        parser.parse()
+    }
+    
+    // MARK: - XMLParserDelegate
+    
+    func parser(_ parser: XMLParser, didStartElement elementName: String, namespaceURI: String?, qualifiedName qName: String?, attributes attributeDict: [String : String] = [:]) {
+        if elementName == "epgItem" {
```

Comment:
All strings should be extracted to constants, especially `playoutItem` as it is used more than one time